brave/brave-core
# Migrated brave theme prefs from profile to local state

With this, we can fix brave theme related bugs with multi profile
because all profiles use browser-wide prefs.
So far, system theme could be changed whenever new profile is used
because brave theme was profile prefs but system theme is browser
wide setting.
And did cleanup and refactoring brave theme related implementations.

Refactoring point is separation of theme managing and dark mode managing.
So far, theme service has all dark mode handling logic. Mixing of dark mode and theme concepts
could cause confusion and made difficult to read the code.
I think theme service should handle browser theme based on underlying dark mode status.
So, all dark mode logic is separated from theme service in `dark_mode` namespace.

New `kBraveDarkMode` prefs are introduced and previous prefs are deprecated.
The value of `kBraveThemeType` is migrated to `kBraveDarkMode`.

fix brave/brave-browser#5373
fix brave/brave-browser#5557

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:
1. Open browser and check current theme type is same with previously used type
2. Create new profile and check its new window also has same theme option
3. Check changing option from one of profiles affects all other profile's window.
    All should have same theme type.
 
Refer issues for bug scenarios

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
